### PR TITLE
AAV154/getGraphDiff tests

### PR DESCRIPTION
**Pull request by itdev**

### CODE REVIEW

**Comment by itdev** [comment]
I think that decomposition to several tests with various entry data would be better for readability.

**Comment by Onatolich** [comment]
Then change test names to not mention internal methods there.

File: src/components/Graph/helpers/__specs__/getGraphDiff.spec.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      { id: '1' },
+    ],
+    edges: [
+      {
+        id: '1',
+        source: '1',
+        target: '2',
+      },
+    ],
+  };
```
Comment (Onatolich) [inline src/components/Graph/helpers/__specs__/getGraphDiff.spec.js]: `next` could be just `{...prev}`

File: src/components/Graph/helpers/__specs__/getGraphDiff.spec.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
+import getGraphDiff from '../getGraphDiff';
+
+test('should return "add" nodes and edges', () => {
```
Comment (Onatolich) [inline src/components/Graph/helpers/__specs__/getGraphDiff.spec.js]: It does not return only added nodes. Also you're testing here not only nodes, but edges as well. Proper description will be `new entities ids should be presented in corresponded result add lists`
Same thing about next descriptions

File: src/components/Graph/helpers/__specs__/getGraphDiff.spec.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      },
+    ],
+  };
+  expect(getGraphDiff(next, prev)).toEqual({
+    edges: { add: [], modify: [], remove: ['2'] },
+    nodes: { add: [], modify: [], remove: ['1', '2'] },
+  });
+});
+
+test('should return "modify" nodes and edges if comparableField list value is exist', () => {
```
Comment (Onatolich) [inline src/components/Graph/helpers/__specs__/getGraphDiff.spec.js]: What is comparableField?

File: src/components/Graph/helpers/__specs__/getGraphDiff.spec.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      },
+    ],
+  };
+  expect(getGraphDiff(next, prev)).toEqual({
+    edges: { add: [], modify: ['1'], remove: [] },
+    nodes: { add: [], modify: ['1'], remove: [] },
+  });
+});
+
+test('should return empty nodes and edges if next and prev attributes are equal', () => {
```
Comment (Onatolich) [inline src/components/Graph/helpers/__specs__/getGraphDiff.spec.js]: `should return empty diff if there is no changes between prev and next graphs`

File: src/components/Graph/helpers/__specs__/getGraphDiff.spec.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    ],
+  };
+
+  expect(getGraphDiff(next, prev)).toEqual({
+    edges: { add: [], modify: [], remove: ['2'] },
+    nodes: { add: [], modify: [], remove: ['1', '2'] },
+  });
+});
+
+test('new entities ids should be presented in corresponded result modify lists if comparableField list value is exist', () => {
```
Comment (Onatolich) [inline src/components/Graph/helpers/__specs__/getGraphDiff.spec.js]: This line is too long. Does eslint accept it? But anyway this test summary should be `modified entities ids should be presented in corresponded result modify lists`

File: src/components/Graph/helpers/__specs__/getGraphDiff.spec.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    edges: [],
+  };
+
+  expect(getGraphDiff(next, prev)).toEqual({
+    edges: { add: ['1'], modify: [], remove: [] },
+    nodes: { add: ['1'], modify: [], remove: [] },
+  });
+});
+
+test('new entities ids should be presented in corresponded result remove lists', () => {
```
Comment (Onatolich) [inline src/components/Graph/helpers/__specs__/getGraphDiff.spec.js]: New entities should be presented in remove list?

File: src/components/Graph/helpers/__specs__/getGraphDiff.spec.js (hunk 5 above)
Comment (itdev) [reply src/components/Graph/helpers/__specs__/getGraphDiff.spec.js]: eslint config doesn't have "max-len" parameter. So linter passed such lines with now warnings.

File: src/components/Graph/helpers/__specs__/getGraphDiff.spec.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    edges: [],
+  };
+
+  expect(getGraphDiff(next, prev)).toEqual({
+    edges: { add: ['1'], modify: [], remove: [] },
+    nodes: { add: ['1'], modify: [], remove: [] },
+  });
+});
+
+test('new entities ids should not be presented in corresponded result remove lists', () => {
```
Comment (Onatolich) [inline src/components/Graph/helpers/__specs__/getGraphDiff.spec.js]: Wrong again. `removed entities ids should be presented in corresponded result remove lists`

File: src/components/Graph/helpers/__specs__/getGraphDiff.spec.js (hunk 1 above)
Comment (itdev) [reply src/components/Graph/helpers/__specs__/getGraphDiff.spec.js]: Done for prev = {...next}

**Pull request merged**